### ISSUE TITLE
Query after the skillBot execution is still going to the skillBot rather than the rootBot.

### ISSUE THREAD

**Issue by jemjos** [microsoft/botframework-sdk#6268]
## Version
v4

## Describe the bug
Issue: We have used the skillBot sample to create a bot which handles proactive as well as non-proactive cards messages. While the bot works as intended for the non-proactive usecase i.e., card acquired by interacting with the bot, but when the card is proactive and after some action is done on the adaptive card, the query from the user goes to the skill bot instead of root bot.  
   
Observation:
- We are able to that the skill bot completed the execution in the logs.
- Removing or changing stepContext.next from final step and adding replaceDialog causes app to go to an infinite loop.
- Tried different changes in the code but still the message after completion if skillBot goes back to skillBot itself.
- Also noted, removing the stepContext.next final step affects non-proactive usecase which is working.

## To Reproduce
Sent a proactive card using bot endpoint to team from bot running this code https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/javascript_nodejs/81.skills-skilldialog

## Expected behavior
Once skillbot execution is done, query should go to rootbot.

## Additional context
Bot is tested in Teams. Adaptive cards are being used.

## Tracking Status

**Dotnet SDK** [TODO]()
- [ ] PR
- [ ] Merged

**Javascript SDK** [TODO]()
- [ ] PR
- [ ] Merged

**Python SDK** [TODO]()
- [ ] PR
- [ ] Merged

**Java SDK** [TODO]()
- [ ] PR
- [ ] Merged

**Samples** [TODO]()
- [ ] PR
- [ ] Merged

**Docs** [TODO]()
- [ ] PR
- [ ] Merged

**Tools** [TODO]()
- [ ] PR
- [ ] Merged

**Comment by jemjos** [comment]
[issue-docs.zip](https://github.com/microsoft/botframework-sdk/files/6344745/issue-docs.zip)

**Comment by stevkan** [comment]
@jemjos, I'm unable to repro the issue, at this moment. But, I want to connect with a colleague for an additional test. I will keep you updated.

**Comment by jemjos** [comment]
@stevkan p.s the issue occurs after the card gets updated once as demonstrated in the attached video.

**Comment by stevkan** [comment]
@jemjos, just re-iterating the request made over Teams:

I'm digging into the issue (referenced above) and am having some trouble replicating the calls you are making from Postman. Could post the details of your calls for sending a card and for updating the activity into the issue? Please include the full URL, headers you pass in, and request body. No need to send keys. I can add those in from my end.

**Comment by jemjos** [comment]
@stevkan please find the requested details attached.
#1 REQUEST1-CREATE -Create request details
#2 REQUEST2-UPDATE -Update request details
#3 TOKEN-GENERATION - Token generation details

POINT TO NOTE: After sending the **CREATE** request, update the body for **UPDATE** request with appropriate conversation-id and activity-id by replacing the placeholders before sending the request. You can retrieve the activity-id from the response body on **CREATE** request.

[TOKEN-GENERATION.txt](https://github.com/microsoft/botframework-sdk/files/6427399/TOKEN-GENERATION.txt)
[REQUEST1-CREATE.txt](https://github.com/microsoft/botframework-sdk/files/6427400/REQUEST1-CREATE.txt)
[REQUEST2-UPDATE.txt](https://github.com/microsoft/botframework-sdk/files/6427401/REQUEST2-UPDATE.txt)

**Comment by tdurnford** [comment]
@stevkan any updates?

**Comment by stevkan** [comment]
@jemjos, thank you for your patience while I've been looking into this. I've been trying to run your code plus your request scripts however I'm getting an error in the bot code that I'm working on hunting down. There are a couple items I want to look into where I suspect the issue is. If I'm not successful, I may reach out to you in Teams.

**Comment by jemjos** [comment]
sure @stevkan. Happy to assist with clarification from my end.

**Comment by stevkan** [comment]
@jemjos, I'm also working on connecting with a colleague regarding this issue. Either way, I'll keep you updated.

**Comment by stevkan** [comment]
@jemjos, thank you for your patience. I looked this over and I am unable to repro the issue you are experiencing. At first, I thought I _was_ getting the same or a similar error until I realized I wasn't completely following the steps you outlined. After re-aligning my steps to yours, your bot's code worked for me correctly.

When I type 'begin', I can click the 'greeting' button which updates the card allowing me to then click 'restart' for a new card to be displayed. Similarly, when I send a proactive message, send the PUT to update the card, I can then click the 'greeting' button followed by the 'restart' button for a new card to be displayed.

Here is a demo showing both working. There is a slight delay in the middle as I update the PUT request in Postman.

![TeamsActivityUpdate](https://user-images.githubusercontent.com/18403946/118900778-a23eeb80-b8c6-11eb-923e-863c9ff27260.gif)

Because of this, I'm unsure what might be blocking you or how to help.

**Comment by jemjos** [comment]
@stevkan could you please confirm that this is the flow you followed while checking proactive message usecase:
1. sent message using POST request.
2. update message using PUT request.
3. click on buttons.

If this is the flow, could you please confirm whether any code changes/ package updates where done to the provided code cause  the behavior experience is shared as a video recording initially.

**Comment by stevkan** [comment]
Yes, those are the steps I performed while checking the proactive message usecase. I haven't made any changes to the code.

However, what I don't know is the specific version of the SDK you are using. You only list v4 in the issue description. I haven't change the installed version listed in package.json (v4.9.1). Is that the version you have installed, as well?

**Comment by stevkan** [comment]
@jemjos, would connecting over Teams so you could demo the issue (as I am unable to repro) be of value? I reached out to you over Teams, as well. Let me know and we can set a time up.

**Comment by stevkan** [comment]
Connecting with @jemjos over Teams to review issue on 4/2.

**Comment by stevkan** [comment]
Met with @jemjos over Teams. Witnessed his code producing the error despite everything looking equivalent to my setup which does not produce the error. Sending my code zipped up for him to test with.

**Comment by jemjos** [comment]
@stevkan thanks for sharing the zip back. i tested with the below setup and still facing the same issue,
1. Both bots linked to a respective bot channel registration via ngrok tunneling which was missing previously as we were using only one.
2. Running the code you shared which is working for you.

**Comment by stevkan** [comment]
@jemjos, thank you for your patience. Unfortunately, I was running into some Teams issue with uploading a new app (for testing with just bot channel registrations). But, I was able to find a way around that, finally.

And, yes, when using two bot channel registrations, only, I am able to repro the issue you are experiencing. I'm reaching out to a colleague to see if this is expected behavior with this kind of setup. However, I think the quick, and probably likely fix, is for the main (root) bot to be hosted with a full web app bot vs. only a bot channel registration.

**Comment by xieofxie** [comment]
@jemjos  One off-topic question: are these REST APIs in Postman public available? Thanks
![image](https://user-images.githubusercontent.com/2876650/121620729-6948ff80-ca9d-11eb-97e4-63c6a5236821.png)

**Comment by stevkan** [comment]
@jemjos, thank you for your patience on this. I'm still connecting with a colleague but hope to have an update for you soon.

**Comment by stevkan** [comment]
@jemjos, can you please try your actions, again? In testing your code, again, for meeting with a colleague, I went thru the steps using two bot channel registrations for the appId and password. It worked without issue so I'm curious if it resolved itself for you, as well.

**Comment by jemjos** [comment]
sure @stevkan. will check and update you.

**Comment by jemjos** [comment]
@xieofxie yes, the endpoints are publicly available. you should be able to access them with the appropriate service URL and token.

**Comment by stevkan** [comment]
@jemjos, I'm closing this due to inactivity. If the problem persists, please feel free to reopen.
